linera-core/src/worker.rs

```diff
@@ -977,17 +972,14 @@ where
         if time_till_block > 0 {
             tokio::time::sleep(Duration::from_micros(time_till_block)).await;
         }
-        let (messages, state_hash) = {
-            let (messages, state_hash) = chain.execute_block(block).await?;
-            // Verify that the resulting chain would have no unconfirmed incoming messages.
-            chain.validate_incoming_messages().await?;
-            // Reset all the staged changes as we were only validating things.
-            chain.rollback();
-            (messages, state_hash)
-        };
+        let executed_block = chain.execute_block(proposal.content.block.clone()).await?;
+        // Verify that the resulting chain would have no unconfirmed incoming messages.
+        chain.validate_incoming_messages().await?;
+        // Reset all the staged changes as we were only validating things.
+        chain.rollback();
         // Create the vote and store it in the chain state.
         let manager = chain.manager.get_mut();
-        manager.create_vote(proposal, messages, state_hash, self.key_pair());
+        manager.create_vote(proposal, executed_block, self.key_pair());
```

Comment:
I'm actually not convinced: The current code avoids this `..block.clone()` while the new code has redundant parameters in `create_vote` (that one could argue require an `assert_eq!(proposal.content.block, executed_block.block)`).